### PR TITLE
fetch course front page on course details load

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-13197
affects: Parent
release note: none

Test plan:
in course details:
1. go to a course with a front page set.  Make sure homepage tab is shown but not summary or syllabus.
2. go to a course with a syllabus set as front page.  Make sure syllabus tab and summary loads
3. go to a course with no front page set or syllabus and make sure syllabus or summary tab or homepage do not show.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1013687&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-13197](https://instructure.atlassian.net/browse/MBL-13197)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Parent/Parent/Course/CourseDetailsViewController.swift:41:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/Parent/Course/CourseDetailsViewController.swift:45:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/Parent/Course/CourseDetailsViewController.swift:86:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Parent/Parent/Course/CourseDetailsViewController.swift [hunk 1]
```diff
@@ -79,14 +84,31 @@ class CourseDetailsViewController: HorizontalMenuViewController {
...
+            switch course.defaultView {
+            case .syllabus:
+                if let body = course.syllabusBody, !body.isEmpty {
+                    configureSyllabus()
+                    configureSummary()
+                }
+            case .wiki:
+                if let page = frontPages.first, !page.body.isEmpty {
+                    configureFrontPage()
+                    configureSummary()
```
Comment (armstrongnate) [inline Parent/Parent/Course/CourseDetailsViewController.swift]: We shouldn't show summary here. The summary tab and the syllabus tab go hand-in-hand so it should only be shown when the syllabus is shown.

Comment (bluwave) [reply Parent/Parent/Course/CourseDetailsViewController.swift]: fixed

File: Parent/Parent/Course/CourseDetailsViewController.swift [hunk 2]
```diff
@@ -120,7 +141,14 @@ extension CourseDetailsViewController: HorizontalPagedMenuDelegate {
         case .grades:
             return NSLocalizedString("Grades", comment: "")
         case .syllabus:
-            return NSLocalizedString("Syllabus", comment: "")
+            switch courses.first?.defaultView {
+            case .wiki:
+                return NSLocalizedString("Homepage", comment: "")
```
Comment (armstrongnate) [inline Parent/Parent/Course/CourseDetailsViewController.swift]: Web calls this "Home". Did Dave want it to be "Homepage"?

Comment (armstrongnate) [inline Parent/Parent/Course/CourseDetailsViewController.swift]: Checked with Dave, he says the home page tab should be titled based on the content of the home page. So this should be "Front Page".

**Comment by armstrongnate** [comment]
@bluwave one more thing, checked with Peyton and he says that if a syllabus exists we should always show it. I'll update the ticket with that criteria.

File: Parent/Parent/Course/CourseDetailsViewController.swift (hunk 2 above)
Comment (bluwave) [reply Parent/Parent/Course/CourseDetailsViewController.swift]: Peyton told me to call it Homepage

**Comment by bluwave** [comment]
@armstrongnate - talked with peyton and dave, changed to frontpage, also, peyton said the following
> so we wouldn’t show syllabus if it existed but wasn’t course home

so i think we are good.

**Pull request merged**